pages/DSNP/Graph.md

```diff
@@ -2,29 +2,126 @@
+Relationships between users are the graph edges.
 
-The public graph is anchored in state transitions recorded in [Graph Change Announcements](Types/GraphChange.md).
-Playing all state transitions in [order](Announcements.md#ordering-announcements) will generate the current state of the public graph.
+## Edges
 
-## Non-Normative
+Graph edges that originate from a user are stored as DSNP [User Data](UserData.md) associated with a given user.
 
-### Private Graph
+Each graph edge consists of the following data:
+* DSNP User Id (64-bit unsigned integer)
+  * <mark>To allow for optimal compression, User Ids are stored using the <tt>long</tt> type in Avro schema, which is a 64-bit _signed_ integer.
+    Care should be taken to ensure that User Id values greater than or equal to 2<sup>63</sup>, where used by an implementation, are converted correctly between signed and unsigned representations.</mark>
+* Timestamp when relationship was created, in seconds since the [Unix epoch](https://en.wikipedia.org/wiki/Unix_time)
 
-Implementation coming in the future. See the [DSNP Whitepaper](https://github.com/LibertyDSNP/papers) for foundational ideas.
+GraphEdge objects MUST conform to the following [Avro](https://avro.apache.org) schema:
 
-### Following
+```
+{
+    "namespace": "org.dsnp.userdata",
+    "name": "GraphEdge",
+    "type": "record",
+    "doc": "A relationship to another DSNP user",
+    "fields": [
+        {
+            "name": "userId",
+            "type": "long",
+            "doc": "DSNP User Id of object of relationship"
+        },
+        {
+            "name": "since",
+            "type": "long",
+            "doc": "Timestamp in Unix epoch seconds when this relationship was originally established"
+        }
+    ]
+}
+```
 
-A "follow" is the act of publicly following a [DSNP User Identity](Identifiers.md#dsnp-user-id), which results in adding this DSNP User Id to a user's social graph.
+## User Data Sets
 
-An "unfollow" is the act of publicly unfollowing a [DSNP User Identity](Identifiers.md#dsnp-user-id) which results in the removal of this DSNP User Id from a user's social graph.
+DSNP user data structures capture three distinct sets of relationships, as described below.
 
-### Friendship
+### Public Follows
 
-There is currently no concept of "friendship" within the DSNP network.
-Friendship requires a mutual acknowledgment between two different DSNP User Identities.
-Friendship could be thought of as "mutual following" - where two DSNP User Identities are following each other.
+Each public follow records a unidirectional relationship where one user (the controller of the graph) is interested in following the public speech of another user.
+When follows are public, they are visible to any other user or system, including the user being followed.
+
+The `publicFollows` user data type is composed of a set of GraphEdge objects, which is then compressed.
```

Comment:
Lines like this one feel out of place without additional details here, but those details are over on UserData

Reply:
could always link to "GraphEdge"

Reply:
Link to the "User Data Types" table